app/cdap/components/DirectiveInput/index.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, IconButton } from '@material-ui/core';
+import InputPanel from 'components/DirectiveInput/Components/InputPanel';
+import DirectiveUsage from 'components/DirectiveInput/Components/DirectiveUsage';
+import {
+  IDirectiveInputProps,
+  IDirectivesList,
+  IDirectiveUsage,
+} from 'components/DirectiveInput/types';
+import { formatDirectiveUsageData, handlePasteDirective } from 'components/DirectiveInput/utils';
+import T from 'i18n-react';
+import React, { useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+import { PREFIX, multipleColumnDirective } from 'components/DirectiveInput/constants';
+import { grey } from '@material-ui/core/colors';
+import CloseOutlinedIcon from '@material-ui/icons/CloseOutlined';
+
+const SimpleBox = styled(Box)`
+  display: block;
+  box-shadow: -3px -4px 15px rgba(68, 132, 245, 0.25);
+`;
+
+const SearchBarWrapper = styled(Box)`
+  padding: 10px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const InputWrapper = styled(Box)`
+  width: 100%;
+  display: flex;
+  align-items: center;
+`;
+
+const DirectiveUsageWrapper = styled(Box)`
+  background: ${grey[700]};
+`;
+
+const InputComponent = styled.input`
+  width: 95%;
+  margin-left: 5px;
+  outline: 0;
+  border: 0;
+  background: transparent;
+  color: #ffffff;
+`;
+
+const LabelComponent = styled.label`
+  color: #94ec98;
+  font-size: 14px;
+  margin-bottom: 0;
+`;
+
+const DirectiveBox = styled(Box)`
+  background-color: #ffffff;
+  position: fixed;
+  bottom: 50px;
+  width: 100%;
+`;
+
+const CloseIconButton = styled(IconButton)`
+  color: #ffffff;
+`;
+
+export default function ({
+  columnNamesList,
+  onDirectiveInputHandler,
+  onClose,
+  openDirectivePanel,
+}: IDirectiveInputProps) {
+  const [inputBoxValue, setInputBoxValue] = useState<string>('');
```

Comment:
if you have more than 4 useStates you should useDispatch instead

Reply:
sorry i meant useReducer

Reply:
Fixed